packages/neon/neon/lib/src/widgets/user_avatar.dart

```diff
@@ -0,0 +1,199 @@
+                  return widget.account.client.core.getAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        CircleAvatar(
+          radius: widget.size / 2,
+          child: ClipOval(
+            child: NeonCachedApiImage(
+              account: widget.account,
+              cacheKey: 'avatar-$effectiveUsername-${isDark ? 'dark' : 'light'}$size',
+              download: () async {
+                if (isDark) {
+                  return widget.account.client.core.getDarkAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                } else {
+                  return widget.account.client.core.getAvatar(
+                    userId: effectiveUsername,
+                    size: size,
+                  );
+                }
+              },
+            ),
+          ),
+        ),
+        if (widget.showStatus) ...[
+          if (widget.username != null) ...[
+            ResultBuilder<NextcloudUserStatusPublicStatus?>(
+              stream: _userStatusBloc.otherUserStatuses.map(
+                (final statuses) => statuses[effectiveUsername] ?? Result.loading(),
+              ),
+              builder: (final context, final userStatus) => _userStatusIcon(
+                loading: userStatus.loading,
+                error: userStatus.error,
+                userStatus: userStatus.data != null
+                    ? _UserStatus(
+                        icon: userStatus.data!.icon,
+                        status: userStatus.data!.status,
+                      )
+                    : null,
+              ),
+            ),
+          ] else ...[
+            ResultBuilder<NextcloudUserStatusStatus?>(
+              stream: _userStatusBloc.userStatus,
+              builder: (final context, final userStatus) => _userStatusIcon(
+                loading: userStatus.loading,
+                error: userStatus.error,
+                userStatus: userStatus.data != null
+                    ? _UserStatus(
+                        icon: userStatus.data!.icon,
+                        status: userStatus.data!.status,
+                      )
+                    : null,
+              ),
+            ),
+          ],
+        ],
+      ],
+    );
+  }
+
+  Widget _userStatusIcon({
+    required final bool loading,
+    required final Object? error,
+    required final _UserStatus? userStatus,
+  }) {
+    final hasEmoji = userStatus?.icon != null;
+    final factor = hasEmoji ? 2 : 3;
+    return SizedBox(
+      height: widget.size,
+      width: widget.size,
+      child: Align(
+        alignment: Alignment.bottomRight,
+        child: Container(
+          height: widget.size / factor,
+          width: widget.size / factor,
+          decoration: loading || error != null || userStatus == null || hasEmoji
```

Comment:
What is this checking?
I have no idea; we should extarnalize this into a meaningful variable